invenio_rdm_migrator/streams/actions/users.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""User actions module."""
+
+from dataclasses import dataclass
+from typing import Optional
+
+from ...actions import LoadAction, LoadData
+from ...load.postgresql.transactions.operations import Operation, OperationType
+from ...transform import EncryptMixin
+from ..models.users import LoginInformation, SessionActivity, User, UserProfile
+
+# GLOBAL FIXME: partial updates would allow to remove the encrypt mixin from many of the
+# actions, since only user registration and edit (with password change should) do the
+# re-encryption. Would save time too.
+
+
+@dataclass
+class UserData(LoadData):
+    """User action data."""
+
+    user: dict
+    login_information: Optional[dict] = None
+    sessions: Optional[list] = None
+
+
+class UserRegistrationAction(LoadAction, EncryptMixin):
+    """Registers a user."""
+
+    name = "register-user"
+    data_cls = UserData
+
+    def __init__(self, data, **kwargs):
+        """Constructor."""
+        # Explicit calls, otherwise MRO only initializes LoadAction
+        LoadAction.__init__(self, data, **kwargs)
+        EncryptMixin.__init__(self)
+
+    def _generate_rows(self, **kwargs):
+        """Generates rows for a new user."""
+        # https://github.com/inveniosoftware/invenio-rdm-migrator/issues/123
+        from datetime import datetime
+
+        self.data.user["created"] = datetime.fromtimestamp(
+            self.data.user["created"] / 1_000_000
+        )
+        self.data.user["updated"] = datetime.fromtimestamp(
+            self.data.user["updated"] / 1_000_000
+        )
```

Comment:
IIRC we discussed that maybe we could change this in debezium config. That's the linked issue on the comment above. Otherwise, yes, it should de done in transform and deduplicate all code.

Reply:
So in our DB the types are `timestamp without time zone`. We might be able to tweak the configuration https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-temporal-types using mode `connect` and the kafka transformer https://docs.confluent.io/platform/current/connect/transforms/timestampconverter.html. Not sure if we want to delegate that to kafka.